src/neighborhoods/metricball.jl

```diff
@@ -36,40 +36,29 @@ struct MetricBall{R,M} <: Neighborhood
   metric::M
 end
 
-function MetricBall(radii::SVector{Dim,T}, rotation=nothing) where {Dim,T}
-  # default rotation
-  rot = if isnothing(rotation)
-    if Dim == 2
-      ClockwiseAngle(zero(T))
-    elseif Dim == 3
-      EulerAngles(zeros(T, Dim)...)
-    else
-      throw(ErrorException("not implemented"))
-    end
-  else
-    rotation
-  end
+function _default_rotation(::Val{2}, T)
+  one(Angle2d{T})
+end
+
+function _default_rotation(::Val{3}, T)
+  one(QuatRotation{T})
+end
```

Comment:
The default rotation can be obtained from `_default_rotation` function. The error not for 2 or 3-dimensional rotation is now `MethodError` instead of `ErrorException("not implemented")`.

Reply:
So QuatRotation is the equivalent of EulerAngles in the 3D case?

Reply:
Can you explain why we need to wrap the rotation type in `one`?

Reply:
>So QuatRotation is the equivalent of EulerAngles in the 3D case?

Not equivalent. Rotations with quaternions are more useful than Euler angles in many cases. See [gimbal lock](https://en.wikipedia.org/wiki/Gimbal_lock) for more information.